chimney/src/test/scala/io/scalaland/chimney/PartialTransformerSumTypeSpec.scala

```diff
@@ -0,0 +1,312 @@
+
+      (short.Zero: short.NumScale[Int, Nothing])
+        .transformIntoPartial[long.NumScale[String]]
+        .asOption ==> Some(long.Zero)
+      (short.Million(4): short.NumScale[Int, Nothing])
+        .transformIntoPartial[long.NumScale[String]]
+        .asOption ==> Some(long.Million("4"))
+      (short.Billion(2): short.NumScale[Int, Nothing])
+        .transformIntoPartial[long.NumScale[String]]
+        .asOption ==> Some(long.Milliard("2"))
+      (short.Trillion(100): short.NumScale[Int, Nothing])
+        .transformIntoPartial[long.NumScale[String]]
+        .asOption ==> Some(long.Billion("100"))
+    }
+
+    test(
+      """transforming flat hierarchies from "subset" of case classes to "superset" of case classes without modifiers when common corresponding types are transformable with Partial Transformer"""
+    ) {
+      implicit val intToDoubleTransformer: PartialTransformer[Int, Double] =
+        (a: Int, _) => partial.Result.fromValue(a.toDouble)
+
+      (shapes1.Triangle(shapes1.Point(0, 0), shapes1.Point(2, 2), shapes1.Point(2, 0)): shapes1.Shape)
+        .transformIntoPartial[shapes3.Shape]
+        .asOption ==>
+        Some(shapes3.Triangle(shapes3.Point(2.0, 0.0), shapes3.Point(2.0, 2.0), shapes3.Point(0.0, 0.0)))
+      (shapes1.Rectangle(shapes1.Point(0, 0), shapes1.Point(6, 4)): shapes1.Shape)
+        .transformIntoPartial[shapes3.Shape]
+        .asOption ==>
+        Some(shapes3.Rectangle(shapes3.Point(0.0, 0.0), shapes3.Point(6.0, 4.0)))
+
+      implicit val intParserOpt: PartialTransformer[String, Int] =
+        PartialTransformer(_.parseInt.toPartialTransformerResult)
+      import numbers._, ScalesPartialTransformer.shortToLongPartialInner
+
+      (short.Zero: short.NumScale[String, Nothing])
+        .transformIntoPartial[long.NumScale[Int]]
+        .asOption ==> Some(long.Zero)
+      (short.Million("4"): short.NumScale[String, Nothing])
+        .transformIntoPartial[long.NumScale[Int]]
+        .asOption ==> Some(long.Million(4))
+      (short.Billion("2"): short.NumScale[String, Nothing])
+        .transformIntoPartial[long.NumScale[Int]]
+        .asOption ==> Some(long.Milliard(2))
+      (short.Trillion("100"): short.NumScale[String, Nothing])
+        .transformIntoPartial[long.NumScale[Int]]
+        .asOption ==> Some(long.Billion(100))
+
+      (short.Million("x"): short.NumScale[String, Nothing])
+        .transformIntoPartial[long.NumScale[Int]]
+        .asOption ==> None
+      (short.Billion("x"): short.NumScale[String, Nothing])
+        .transformIntoPartial[long.NumScale[Int]]
+        .asOption ==> None
+      (short.Trillion("x"): short.NumScale[String, Nothing])
+        .transformIntoPartial[long.NumScale[Int]]
+        .asOption ==> None
+    }
+
+    test(
+      """transforming nested sealed hierarchies from "subset" of case classes to "superset" of case classes without modifiers when common corresponding types are transformable"""
+    ) {
+      // TODO: failing
```

Comment:
If it's a bug, can you create a ticket and link to it in the TODO comment?

Reply:
It wasn't actually a bug